tests/unit/test_spaces.cpp

```diff
@@ -230,4 +243,78 @@ TEST_P(FP64SpacesOptimizationTest, FP64InnerProductTest) {
 
 INSTANTIATE_TEST_SUITE_P(FP64OptFuncs, FP64SpacesOptimizationTest, testing::Range(1UL, 8 * 2UL));
 
+class BF16SpacesOptimizationTest : public testing::TestWithParam<size_t> {};
+
+TEST_P(BF16SpacesOptimizationTest, BF16InnerProductTest) {
+    Arch_Optimization optimization = getArchitectureOptimization();
+    size_t dim = GetParam();
+    bfloat16 v[dim];
+    bfloat16 v2[dim];
+    for (size_t i = 0; i < dim; i++) {
+        v[i] = float_to_bf16((float)i);
+        v2[i] = float_to_bf16(((float)i + 1.5));
+    }
+
+    dist_func_t<float> arch_opt_func;
+    float baseline = BF16_InnerProduct_LittleEndian(v, v2, dim);
```

Comment:
Consider adding validation that the baseline equals to FP32 calculation

Reply:
but it shouldn't